Sources/SwiftSyntax/RawSyntaxNodes.swift.gyb

```diff
@@ -59,8 +59,15 @@ public struct Raw${node.name}: Raw${node.name if node.is_base() else node.base_t
 %   end
 %
 %   if node.is_syntax_collection():
-  public init<C: Collection>(elements: C, arena: SyntaxArena)  where C.Element == Raw${node.collection_element_type} {
-    let raw = RawSyntax.makeLayout(kind: .${node.swift_syntax_kind}, from: elements.map { $0.raw }, arena: arena)
+  public init(elements: [Raw${node.collection_element_type}], arena: __shared SyntaxArena) {
```

Comment:
Could we get the same performance improvement by adding
```swift
@_specialize(where C == Array<Raw${node.collection_element_type}>)
```
to the declaration while still having the convenience of being able to pass other collections in?

Reply:
Probably yes. But it would increase the code size. Since most clients should never use these `RawSyntax` APIs, I don't think it's worth the bother.